osf/models/node.py

```diff
@@ -2787,30 +2790,33 @@ def remove_node(self, auth, date=None):
         :param datetime date: `datetime.datetime` or `None`
         """
         # TODO: rename "date" param - it's shadowing a global
-        if not self.can_edit(auth):
+        hierarchy = Node.objects.get_children(self, active=True, include_root=True)
+        is_admin = Contributor.objects.filter(node=OuterRef('pk'), admin=True, user=auth.user)
+        if len(hierarchy.annotate(has_admin_perm=Exists(is_admin)).filter(has_admin_perm=False)):
```

Comment:
Does this intentionally exclude users with `write` perms? Seems like `can_edit` allowed write users but this limits to admin.

Reply:
thanks for pointing this out - I meant to comment here.

I was mainly made permissions stricter due to @brianjgeiger 's comment [here](https://openscience.atlassian.net/browse/PLAT-1127?focusedCommentId=70414&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-70414), saying you needed admin perms through the whole hierarchy to be able to delete.

It also looks to me like we currently have some inconsistencies around this.  Model-level and api currently allow write-users to delete, but legacy UI requires you to be an admin.